yt_dlp/extractor/cbc.py

```diff
@@ -390,7 +390,13 @@ def _real_extract(self, url):
         show = match.group('show')
         show_info = self._download_json(self._API_BASE + show, season_id)
         season = int(match.group('season'))
-        season_info = try_get(show_info, lambda x: x['seasons'][season - 1])
+
+        # Search for the season with the right number
+        season_info = None
+        for show_season in show_info["seasons"]:
+            if show_season.get("season") == season:
+                season_info = show_season
+                break
```

Comment:
```suggestion
        season_info = next((s for s in show_info['seasons'] if s.get('season') == season), None)
```

Reply:
You've made a similar suggestion using `next` before for my PRs. I find these kinds of rewrites to be much less readable then the original code, and you don't gain anything other than having it take up fewer lines. This goes against the [Zen of Python](https://www.python.org/dev/peps/pep-0020/#the-zen-of-python) in my opinion.

I'll commit the change because it's not a big deal, I want this PR in, and I really appreciate all the work you've done to maintain and grow this repo. But I'm confused by these sorts of suggestions.

Reply:
I think it's exactly the reverse. Setting to None, then modifying and breaking inside the loop is a C programmers way of doing things. Using `next` is the pythonic way to get the first element that satisfies a condition.

The code even reads like exactly what we want - "get the first (next) item (s) from (for) show_info.seasons where (if) s.season is season"; as opposed to your code where we need to explicitly reason about the control flow

If the condition was a function, `next(filter(func, show_info['seasons']))` would be even better. But having to construct a lambda makes this less elegant in this instance

Reply:
I see your point, perhaps I'm just too used to one way of doing things. Thanks for taking the time to explain this. Let me know if anything else is needed for my PR.